### ISSUE TITLE
Upgrade qulice to 0.16.2

### ISSUE THREAD

**Issue by krzyk** [yegor256/netbout#1013]
Netbout is currently at version 0.15.2, while 0.16.2 was released with following new fixes:
* teamed/qulice#594 Implement check to prohibit Javadoc in methods with '@Override'
* teamed/qulice#634 Qulice doesn't check order of Javadoc tags
* teamed/qulice#627 Qulice allows catch parameters not following naming convention
* teamed/qulice#398 DependenciesValidator should support excludes
* teamed/qulice#646 Prohibit usage of plain JUnit assertions
* teamed/qulice#144 prohibit declaring a checked exception if it's not thrown
* teamed/qulice#666 NonStaticMethodCheck should ignore IT tests and small fixes
* teamed/qulice#411 IndentationCheck is wrong for annotaions
* teamed/qulice#623 NonStaticMethodCheck complains about methods that just throw an exception
* teamed/qulice#676 Qulice fails with NullPointerException (fix for dependencies excludes)
* teamed/qulice#655 All usages of apache commons CharEncoding.UTF_8 should be prohibited
* teamed/qulice#660 Check to make sure that abbreviations are written in camel case
* teamed/qulice#656 Prohibit empty line at the start and at the end of javadoc
* teamed/qulice#675 Fields should be marked transient only if class is Serializable
* teamed/qulice#655 All usages of apache commons CharEncoding.UTF_8 should be prohibited
* teamed/qulice#625 Ignore LineLength check on lines containing links
* teamed/qulice#502 String concatenation check sometimes doesn't work
* teamed/qulice#653 Empty line between methods of an Anonymous class makes qulice complain

**Comment by dmzaytsev** [comment]
@krzyk advertising ? :)
Thanks, we should actually do that

**Comment by dmzaytsev** [comment]
@karato valid bug

**Comment by krzyk** [comment]
@dmzaytsev yeah, and loosing potential review points :)

**Comment by dmzaytsev** [comment]
@krzyk me, too :)

**Comment by karato** [comment]
> @karato valid bug

@dmzaytsev I tagged this as "bug"

**Comment by karato** [comment]
@krzyk attached this issue to milestone "3.1" (let me know if this is wrong)

**Comment by karato** [comment]
@krzyk thanks for reporting! I topped your account for 15 mins, transaction 77335817

**Comment by kujtimiihoxha** [comment]
@dmzaytsev I could do this .

**Comment by dmzaytsev** [comment]
@karato assign @kujtimiihoxha please

**Comment by karato** [comment]
> @karato assign @kujtimiihoxha please

@dmzaytsev yep, done. @kujtimiihoxha it is your task now

**Comment by kujtimiihoxha** [comment]
@dmzaytsev  I am getting some checkstyle error in some files where there are puzzled written over ```@Override```  should I add this files to the excluded files ?

**Comment by dmzaytsev** [comment]
@krzyk please support @kujtimiihoxha on this question
I think the puzzles should be rewritten a different way

**Comment by krzyk** [comment]
@kujtimiihoxha puzzles above override should be written using double slash comments `//` not Javadoc ones

**Comment by kujtimiihoxha** [comment]
@krzyk Ok I will fix that now .

**Comment by kujtimiihoxha** [comment]
@krzyk  @dmzaytsev  I am getting a lot of findbug violations 
```
 >> M C BED: Non derivable method com.netbout.mock.MkFriends.invite(String) declares throwing an exception that isn't thrown  At MkFriends.java:[line 80]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkFriends.kick(String) declares throwing an exception that isn't thrown  At MkFriends.java:[line 104]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkFriends.iterate() declares throwing an exception that isn't thrown  At MkFriends.java:[line 117]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkFriend.photo() declares throwing an exception that isn't thrown  At MkFriend.java:[line 81]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkFriend.email() declares throwing an exception that isn't thrown  At MkFriend.java:[line 95]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkMessage.date() declares throwing an exception that isn't thrown  At MkMessage.java:[line 91]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkMessage.text() declares throwing an exception that isn't thrown  At MkMessage.java:[line 111]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkMessage.author() declares throwing an exception that isn't thrown  At MkMessage.java:[line 124]
[INFO] >> M C BED: Non derivable method com.netbout.mock.TouchBout.act() declares throwing an exception that isn't thrown  At TouchBout.java:[line 76]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkAttachment.ctype() declares throwing an exception that isn't thrown  At MkAttachment.java:[line 91]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkAttachment.etag() declares throwing an exception that isn't thrown  At MkAttachment.java:[line 104]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkAttachment.read() declares throwing an exception that isn't thrown  At MkAttachment.java:[line 122]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkAttachment.write(InputStream, String, String) declares throwing an exception that isn't thrown  At MkAttachment.java:[line 140]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkMessages.post(String) declares throwing an exception that isn't thrown  At MkMessages.java:[line 90]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkMessages.iterate() declares throwing an exception that isn't thrown  At MkMessages.java:[line 116]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkMessages.search(String) declares throwing an exception that isn't thrown  At MkMessages.java:[line 146]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkAttachments.unseen() declares throwing an exception that isn't thrown  At MkAttachments.java:[line 89]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkAttachments.create(String) declares throwing an exception that isn't thrown  At MkAttachments.java:[line 95]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkAttachments.delete(String) declares throwing an exception that isn't thrown  At MkAttachments.java:[line 112]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkAttachments.iterate() declares throwing an exception that isn't thrown  At MkAttachments.java:[line 130]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkBout.date() declares throwing an exception that isn't thrown  At MkBout.java:[line 94]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkBout.updated() declares throwing an exception that isn't thrown  At MkBout.java:[line 106]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkBout.title() declares throwing an exception that isn't thrown  At MkBout.java:[line 118]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkBout.rename(String) declares throwing an exception that isn't thrown  At MkBout.java:[line 130]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkBout.subscription() declares throwing an exception that isn't thrown  At MkBout.java:[line 143]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkBout.subscription(String) declares throwing an exception that isn't thrown  At MkBout.java:[line 149]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkBout.subscribe(boolean) declares throwing an exception that isn't thrown  At MkBout.java:[line 163]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkInbox.start() declares throwing an exception that isn't thrown  At MkInbox.java:[line 84]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkInbox.bout(long) declares throwing an exception that isn't thrown  At MkInbox.java:[line 104]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkInbox.iterate() declares throwing an exception that isn't thrown  At MkInbox.java:[line 125]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkInbox.search(String) declares throwing an exception that isn't thrown  At MkInbox.java:[line 156]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkAlias.photo() declares throwing an exception that isn't thrown  At MkAlias.java:[line 85]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkAlias.locale() declares throwing an exception that isn't thrown  At MkAlias.java:[line 99]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkAlias.photo(URI) declares throwing an exception that isn't thrown  At MkAlias.java:[line 113]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkAlias.email() declares throwing an exception that isn't thrown  At MkAlias.java:[line 126]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkAlias.email(String) declares throwing an exception that isn't thrown  At MkAlias.java:[line 138]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkAlias.email(String, String, Bout) declares throwing an exception that isn't thrown  At MkAlias.java:[line 151]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkAlias.email(String, String) declares throwing an exception that isn't thrown  At MkAlias.java:[line 157]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkAlias.inbox() declares throwing an exception that isn't thrown  At MkAlias.java:[line 162]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkAliases.check(String) declares throwing an exception that isn't thrown  At MkAliases.java:[line 83]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkAliases.add(String) declares throwing an exception that isn't thrown  At MkAliases.java:[line 102]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkAliases.iterate() declares throwing an exception that isn't thrown  At MkAliases.java:[line 120]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkUser.friends(String) declares throwing an exception that isn't thrown  At MkUser.java:[line 89]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkBase.randomAlias() declares throwing an exception that isn't thrown  At MkBase.java:[line 99]
[INFO] >> M C BED: Non derivable method com.netbout.mock.MkBase.randomBout() declares throwing an exception that isn't thrown  At MkBase.java:[line 124]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkFriends.invite(String) declares throwing an exception that isn't thrown  At MkFriends.java:[line 80]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkFriends.kick(String) declares throwing an exception that isn't thrown  At MkFriends.java:[line 104]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkFriends.iterate() declares throwing an exception that isn't thrown  At MkFriends.java:[line 117]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkFriend.photo() declares throwing an exception that isn't thrown  At MkFriend.java:[line 81]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkFriend.email() declares throwing an exception that isn't thrown  At MkFriend.java:[line 95]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkMessage.date() declares throwing an exception that isn't thrown  At MkMessage.java:[line 91]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkMessage.text() declares throwing an exception that isn't thrown  At MkMessage.java:[line 111]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkMessage.author() declares throwing an exception that isn't thrown  At MkMessage.java:[line 124]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.TouchBout.act() declares throwing an exception that isn't thrown  At TouchBout.java:[line 76]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkAttachment.ctype() declares throwing an exception that isn't thrown  At MkAttachment.java:[line 91]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkAttachment.etag() declares throwing an exception that isn't thrown  At MkAttachment.java:[line 104]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkAttachment.read() declares throwing an exception that isn't thrown  At MkAttachment.java:[line 122]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkAttachment.write(InputStream, String, String) declares throwing an exception that isn't thrown  At MkAttachment.java:[line 140]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkMessages.post(String) declares throwing an exception that isn't thrown  At MkMessages.java:[line 90]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkMessages.iterate() declares throwing an exception that isn't thrown  At MkMessages.java:[line 116]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkMessages.search(String) declares throwing an exception that isn't thrown  At MkMessages.java:[line 146]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkAttachments.unseen() declares throwing an exception that isn't thrown  At MkAttachments.java:[line 89]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkAttachments.create(String) declares throwing an exception that isn't thrown  At MkAttachments.java:[line 95]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkAttachments.delete(String) declares throwing an exception that isn't thrown  At MkAttachments.java:[line 112]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkAttachments.iterate() declares throwing an exception that isn't thrown  At MkAttachments.java:[line 130]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkBout.date() declares throwing an exception that isn't thrown  At MkBout.java:[line 94]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkBout.updated() declares throwing an exception that isn't thrown  At MkBout.java:[line 106]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkBout.title() declares throwing an exception that isn't thrown  At MkBout.java:[line 118]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkBout.rename(String) declares throwing an exception that isn't thrown  At MkBout.java:[line 130]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkBout.subscription() declares throwing an exception that isn't thrown  At MkBout.java:[line 143]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkBout.subscription(String) declares throwing an exception that isn't thrown  At MkBout.java:[line 149]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkBout.subscribe(boolean) declares throwing an exception that isn't thrown  At MkBout.java:[line 163]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkInbox.start() declares throwing an exception that isn't thrown  At MkInbox.java:[line 84]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkInbox.bout(long) declares throwing an exception that isn't thrown  At MkInbox.java:[line 104]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkInbox.iterate() declares throwing an exception that isn't thrown  At MkInbox.java:[line 125]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkInbox.search(String) declares throwing an exception that isn't thrown  At MkInbox.java:[line 156]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkAlias.photo() declares throwing an exception that isn't thrown  At MkAlias.java:[line 85]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkAlias.locale() declares throwing an exception that isn't thrown  At MkAlias.java:[line 99]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkAlias.photo(URI) declares throwing an exception that isn't thrown  At MkAlias.java:[line 113]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkAlias.email() declares throwing an exception that isn't thrown  At MkAlias.java:[line 126]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkAlias.email(String) declares throwing an exception that isn't thrown  At MkAlias.java:[line 138]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkAlias.email(String, String, Bout) declares throwing an exception that isn't thrown  At MkAlias.java:[line 151]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkAlias.email(String, String) declares throwing an exception that isn't thrown  At MkAlias.java:[line 157]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkAlias.inbox() declares throwing an exception that isn't thrown  At MkAlias.java:[line 162]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkAliases.check(String) declares throwing an exception that isn't thrown  At MkAliases.java:[line 83]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkAliases.add(String) declares throwing an exception that isn't thrown  At MkAliases.java:[line 102]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkAliases.iterate() declares throwing an exception that isn't thrown  At MkAliases.java:[line 120]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkUser.friends(String) declares throwing an exception that isn't thrown  At MkUser.java:[line 89]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkBase.randomAlias() declares throwing an exception that isn't thrown  At MkBase.java:[line 99]
[WARNING] FindBugs: M C BED: Non derivable method com.netbout.mock.MkBase.randomBout() declares throwing an exception that isn't thrown  At MkBase.java:[line 124]
```
45 in total, do I have to fix all of them or exclude findbug check from netbout.client

**Comment by krzyk** [comment]
@kujtimiihoxha you should fix as much as budget allows and puzzle with exclude the rest.

**Comment by kujtimiihoxha** [comment]
@krzyk @dmzaytsev  I can see that it is already excluded but it does not exclude this files https://github.com/yegor256/netbout/blob/master/netbout-client/pom.xml#L191

**Comment by krzyk** [comment]
@kujtimiihoxha also there is a bug, that gives false positives for this findbugs errors if method or class is annotated with jcabi aspects annitations

**Comment by kujtimiihoxha** [comment]
@krzyk  the excluded files are in com.netbout.client, i am getting violations in com.netbout.mock my fault sorry.

**Comment by kujtimiihoxha** [comment]
@krzyk  is there an issue I can relate to in about the bug with jcabi annotations so i can mention it in the puzzle?

**Comment by krzyk** [comment]
@kujtimiihoxha https://github.com/teamed/qulice/issues/706, it will be reopened

**Comment by kujtimiihoxha** [comment]
@karato , @krzyk  See PR #1017.

**Comment by karato** [comment]
> @karato , @krzyk  See PR #1017.

@kujtimiihoxha thanks, I'll take a look

**Comment by kujtimiihoxha** [comment]
@krzyk the changes are merged please close this issue.

**Comment by krzyk** [comment]
@kujtimiihoxha thanks

**Comment by karato** [comment]
@krzyk the code made here contains a puzzle `1013-d9e5eb87`/#1028, which will be resolved soon

**Comment by karato** [comment]
@ypshenychka review this ticket please, for compliance with our [quality rules](http://at.teamed.io/qa.html)

**Comment by ypshenychka** [comment]
@karato Quality is good.

**Comment by karato** [comment]
> @karato Quality is good.

@ypshenychka thank you

**Comment by karato** [comment]
@kujtimiihoxha 10 mins was added to the account of @ypshenychka (for QA review), in transaction 78181778. Thanks for your contribution, **38 mins** was added to your account, payment ID is `AP-5F8122040S4962104`, 57 hours and 45 mins spent total. the bonus for fast delivery (m=3465). +38 to your rating, your total score is [+323](http://www.netbout.com/b/36323?open=rating)

**Comment by kujtimiihoxha** [comment]
@karato the task took 24 hours and 20 min, from the moment I was assigned until the moment it was closed. please check the time again.

**Comment by karato** [comment]
> @karato the task took 24 hours and 20 min, from the moment I was assigned until the moment it was closed. please check the time again. 

@kujtimiihoxha see http://at.teamed.io/policy.html#15

**Comment by karato** [comment]
@krzyk last puzzled solved `1013-d9e5eb87`/#1028

**Comment by 0pdd** [comment]
@krzyk the only puzzle [#unknown]() is solved here.
